tests/unit/field/test_text_field.py

```diff
@@ -134,6 +136,60 @@ def test_build_vocab_empty():
     assert field.vocab == vocab
 
 
+def test_build_vocab_setup_all_embeddings():
+    """
+    This test shows that all fields in the embeddings will be included.
+
+    In embeddings and data:
+        blue
+        green
+        yellow
+    In embeddings only:
+        purple
+        gold
+    In data only:
+        white
+
+    Expected vocab:
+        blue
+        green
+        yellow
+        purple
+        gold
+        white
+    """
+
+    field = TextField(
+        embeddings='not-a-real-embedding',
+        setup_all_embeddings=True,
+    )
+
+    embedding_model = MagicMock()
+    embedding_model.vocab = {
+        'blue': torch.rand(5),
+        'green': torch.rand(5),
+        'yellow': torch.rand(5),
+        'purple': torch.rand(5),
+        'gold': torch.rand(5),
+    }
+    embedding_model.__getitem__.side_effect = embedding_model.vocab.__getitem__
+    embedding_model.__contains__.side_effect = embedding_model.vocab.__contains__
+    field._get_embeddings = MagicMock(
+        return_value=(embedding_model, [torch.rand(5)]),
+    )
```

Comment:
I don't think we should mock this method. The embeddings matrix can be provided in the constructor and we can deal with this test without using `_get_embeeddings` method right?

Reply:
Sure but I'd like to keep the API simpler if possible

Reply:
will still make the suggested change though